### ISSUE TITLE
Release 0.9.90

### ISSUE THREAD

**Issue by mttstt** [perak/kitchen-site#396]
Have you some examples how to use "variables" and "update_rule" implemented in the last version of meteor-kitchen?
Thanks

**Comment by perak** [comment]
@mttstt 

**Example for "update_rule" (in .json) : **

```javascript
"update_rule": "doc.status == \"draft\""
```

In GUI it looks like this:

![screen shot 2017-11-27 at 9 51 50 am](https://user-images.githubusercontent.com/5630814/33258130-dd0101d8-d358-11e7-9192-1aff9ad06cd2.png)

**Example for query "variables" **

Two queries are shown, second one ("available_tribes") is using result of the first one ("users_membership") :

```javascript
			{
				"name": "users_membership",
				"collection": "membership",
				"filter": "{\"userId\":\"Meteor.userId()\"}",
				"options": "{\"sort\":{\"createdAt\":1}}"
			},
			{
				"name": "available_tribes",
				"collection": "tribes",
				"filter": "{\"_id\":{\"$nin\":\"%joinedTribeIds\"}}",
				"options": "{\"sort\":{\"name\":1}}",
				"variables": [
					{
						"name": "joinedTribeIds",
						"value": "tribeId",
						"query_name": "users_membership"
					}
				]
			}
```

In GUI it looks like this:

![screen shot 2017-11-27 at 10 05 59 am](https://user-images.githubusercontent.com/5630814/33258678-a838300a-d35a-11e7-8f17-27b48c1a11c6.png)

![screen shot 2017-11-27 at 10 06 06 am](https://user-images.githubusercontent.com/5630814/33258684-af22cb28-d35a-11e7-9711-4db311e4aee8.png)

![screen shot 2017-11-27 at 10 06 15 am](https://user-images.githubusercontent.com/5630814/33258690-b601962c-d35a-11e7-9439-5ef5223f1d92.png)

**Comment by mttstt** [comment]
Is it a Bug ?

1) first query
![image](https://user-images.githubusercontent.com/17787853/45026609-1e257680-b03f-11e8-8631-829e7f9d3dc4.png)


![image](https://user-images.githubusercontent.com/17787853/45027161-ec151400-b040-11e8-8f9b-b9773df933e5.png)


2) query
![image](https://user-images.githubusercontent.com/17787853/45026683-59c04080-b03f-11e8-8758-f4276219ef77.png)


3) Resulting code:
Meteor.publish("activities_list", function() {
var varname = .findOne({_id:this.userId}, {})..findOne({_id:this.userId}, {});
return Activities.find({auditors:varname}, {});
});

**Comment by perak** [comment]
@mttstt what is your `meteor-kitchen --version` ?

This is what I have (v0.9.104) :
```javascript
Meteor.publish("activities_list", function() {
	var varname = Users.findOne({_id:this.userId}, {}).Users.findOne({_id:this.userId}, {});
	return Activities.find({auditors:varname}, {});
});
```

**Comment by perak** [comment]
(v0.9.104 also produces wrong code, but differs from what you reported)

**Comment by perak** [comment]
Fixed v0.9.104 bug (will be in 0.9.105), now I got:
```javascript
Meteor.publish("activities_list", function() {
	var varname = Users.findOne({_id:this.userId}, {})["profile.name"];
	return Activities.find({auditors:varname}, {});
});
```

**Comment by perak** [comment]
@mttstt P.S. in this specific case (because you need data from current user) you can do the same thing without variables/external query, so instead query:

```javascript
{
  "auditors": "%varname"
}
```

You can do simply:
```javascript
{
  "auditors": "Meteor.user()['profile.name']"
}
```

That will produce:

At server:
```javascript
Meteor.publish("activities_list", function() {
	var currentUser = Meteor.users.findOne({ _id: this.userId });
	return Activities.find({auditors:currentUser['profile.name']}, {});
});
```

At client:
```javascript
Activities.find({ auditors: Meteor.user()['profile.name'] })
```

**Comment by perak** [comment]
@mttstt after fixing more issues, this is final code generated with next version `0.9.105`:

```javascript
Meteor.publish("activities_list", function() {
	var varnameObj = Users.findOne({_id:this.userId}, {});
	var varname = objectUtils.getPropertyValue("profile.name", varnameObj);
	return Activities.find({auditors:varname}, {});
});
```

**Comment by mttstt** [comment]
I had the Release 0.9.103: the last release published in the https://www.meteorkitchen.com/news site.

However, thanks very much!

**Comment by perak** [comment]
@mttstt Published 0.9.105
